subsys/usb/class/audio/Kconfig

```diff
@@ -10,4 +10,265 @@ config USB_DEVICE_AUDIO
 
 if USB_DEVICE_AUDIO
 
+config USB_AUDIO_DEVICE_NAME
+	string "USB AUDIO device name template"
+	default "AUDIO"
+	help
+	  Device name template for the AUDIO Devices.
+	  Headphones device would have
+	  name $(USB_AUDIO_DEVICE_NAME)_HEADPHONES.
+	  Microphone device would have
+	  name $(USB_AUDIO_DEVICE_NAME)_MICROPHONE.
+
+module = USB_AUDIO
+module-str = USB Audio
+source "subsys/logging/Kconfig.template.log_config"
+
+menuconfig USB_AUDIO_HEADPHONES
```

Comment:
Would it make sense to split it in Kconfig, Kconfig.headphones and Kconfig.micro?

Reply:
I was thinking about putting all configuration options to *.overlay with relevant binding.

This PR doesn't support configuring 2 pairs of headphones/microphone for now, but I will add it for sure.. This will require having independent configuration options for each pair of Hp/Mic. Having configuration options in Kconfig will require to have fixed number of headphones predefined because Kconfig is not that flexible so it could add configs on the fly. To avoid this I think creating a binding for hp/mic will be nice.

Inside *.overlay user will define as many hps, mics as he wants and the macros needed for configuration will be created automatically.

Reply:
>Having configuration options in Kconfig will require to have fixed number of headphones predefined because Kconfig is not that flexible so it could add configs on the fly.

I guess something like [_subsys/usb/class/Kconfig.template.composite_device_number_](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/usb/class/Kconfig#L30) would help you, please check how it is used for CDC ACM.

Reply:
This is not about *_DEVICE_COUNT. The problem is Kconfig will not generate macros like:
*HP_0_FEATURES_MUTE
*HP_1_FEATURES_MUTE
*HP_2_FEATURES_MUTE
depending on *_DEVICE_COUNT == 3

We need to have configuration symbols created on the fly. This is not possible with Kconfig which needs to have those options defined and user is only enabling them.

In other words Kconfig will not add more options if i specify device count 2 or 3. He needs to know it before it is chosen.

Reply:
You could add a subsys/usb/class/audio/Kconfig.template.audio_hp based on subsys/usb/class/Kconfig.template.composite_device_number, would it not help you?

Reply:
If it could work like that then I would need something like this (below) inside Kconfig. Unfortunately this is not possible.
```
switch (CONFIG_USB_AUDIO_HEADPHONES_DEVICE_COUNT)
case 1:
        module = HEADPONES_0
        default-count = 0
        source "subsys/usb/class/audio/Kconfig.template.audio_hp"
case 2:
        module = HEADPONES_0
        default-count = 0
        source "subsys/usb/class/audio/Kconfig.template.audio_hp"
        module = HEADPONES_1
        default-count = 1
        source "subsys/usb/class/audio/Kconfig.template.audio_hp"
.
.
```
where inside subsys/usb/class/audio/Kconfig.template.audio_hp there would be something like:

```
config USB_AUDIO_$(module)_FEATURES_MUTE
        bool "blah blah"
        default y
        help
          Not working
config USB_AUDIO_$(module)_FEATURES_MUTE
.
.
.
```